src/lib/route-data/route-data-registry.ts

```diff
@@ -24,8 +24,9 @@ export class RouteDataRegistry {
     }
 
     public static getAll(): Readonly<RouteData> {
-        // TODO
-        return null;
+        let object: RouteData = ({ ...this.registry } as unknown) as RouteData;
```

Comment:
enthält das Object dann nicht auch die Methoden der Map??
Der entsprechende Unit-test schlägt auch fehl..

Außerdem hatten wir doch eben besprochen, dass der Type des Rückgabewertes eigentlich `{ [path:string]: Readonly<RouteDataInterface> }` sein müsste!?

Reply:
Der Unit test schlägt fehl weil die register Methode noch nicht implementiert ist ;-) wegen dem Rest muss ich dann schauen wenn die da ist

Reply:
Wegen dem Rückgabewert hatte ich es so verstanden, dass es wichtig ist, dass der RouteDataInterface Teil nicht beschrieben werden darf. Und soweit ich die freeze-Methode verstanden habe, macht die genau das.

Reply:
<img width="686" alt="Bildschirmfoto 2021-01-28 um 14 00 26" src="https://user-images.githubusercontent.com/26869118/106141996-3434b400-6171-11eb-8b91-59438263dbb8.png">
Quelle: https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze#beschreibung

Heißt also, die Werte vom Typ `RouteDataInterface` können so immer noch editiert werden.

Hast du dir mal die TODOs von mir auf dem einen Branch durchgelesen?
Meiner Meinung nach gibt es zwei Möglichkeiten das umzusetzen..

1. Die route data werden eingefroren sobald sie registriert werden. Somit sind sie nicht mehr editierbar. Wenn wir die Map in ein Object konvertieren bleibt dass dann auch so.
2. Wir müssten beim Ausgeben des Objects eine Kopie jeder einzelnen route data erstellen. Diese könnte man dan einfrieren (um dem type Readonly gerecht zu werden), muss man praktisch aber nicht, denn es handelt sich ja um Kopien - die ursprüngliche route data ist so also auch nicht veränderbar.

Reply:
Und wegen dem Unit-Test... implementier halt schnell die registerOne-methode. Habe ich für die get-Methode auch gemacht. Oder füg die Route halt manuell der registry hinzu.

Reply:
BTW: Lösung 1 ist Lösung 2 vorzuziehen, da weniger Speicherverbrauch (es müssen keine unnötigen Kopien erzeugt werden).
Allerdings kann dann eine RouteData eben nicht mehr modifiziert werden nachdem sie registriert wurde.

Reply:
wobei.. tatsächlich könnte man sie einfach verändern und neu registrieren und damit überschreiben..
Damit hat Lösung 1 keine Nachteile und sollte daher umgesetzt werden! Ist tatsächlich auch die, die einfacher umzusetzen ist. Ich werde das dann in die Implementierung der `register` methode mit aufnehmen.